libs/langchain/langchain/vectorstores/pinecone.py

```diff
@@ -303,6 +325,44 @@ def max_marginal_relevance_search(
             embedding, k, fetch_k, lambda_mult, filter, namespace
         )
 
+    @classmethod
+    def get_pinecone(
+        cls,
+        index_name: Optional[str],
+        embedding: Union[Embeddings, Callable],
+        text_key: str,
+        namespace: Optional[str] = None,
+        pool_threads: int = 1,
+        distance_strategy: Optional[DistanceStrategy] = DistanceStrategy.COSINE,
+        **kwargs: Any,
+    ) -> Pinecone:
+        """Get or create a Pinecone index."""
+
+        try:
+            import pinecone
```

Comment:
I've noticed this pattern and I wanted to ask (as I continue ramping up) if it's common to handle imports this way? For example, why not declare the import at the top of the file? Is it mostly a performance concern around importing unused code?

Reply:
Roger - thanks for explaining!